### PR TITLE
RavenDB-12010 - Wrong free disk space for symbolic links in the dashboard

### PR DESCRIPTION
**Pull request by grisha-kotler**
http://issues.hibernatingrhinos.com/issue/RavenDB-12010

### CODE REVIEW

**Comment by ppekrol** [comment]
@grisha-kotler it looks that your changes are killing the runtime. Can you take a look?

**Comment by grisha-kotler** [comment]
done

**Comment by ppekrol** [comment]
Still runtime is failing. Running tests again.

**Comment by grisha-kotler** [comment]
The issue was that passing a StringBuilder with initial capacity and max capacity to the system call didn't work as expected. The system call used the underlying buffer and when it reached the initial capacity limit, we had AccessViolation/StackOverflow.

File: src/Sparrow/Utils/DiskSpaceChecker.cs [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+        public static DiskSpaceResult GetDiskSpaceInfo(string pathToCheck, DriveInfoBase driveInfoBase = null)
+        {
+            if (string.IsNullOrEmpty(pathToCheck))
+                return null;
+
+            if (PlatformDetails.RunningOnPosix)
+            {
+                var statvfs = default(Statvfs);
+                if (Syscall.statvfs(pathToCheck, ref statvfs) != 0)
+                    return null;
```
Comment (ayende) [inline src/Sparrow/Utils/DiskSpaceChecker.cs:35]: Should we log here as well?

File: src/Sparrow/Utils/DiskSpaceChecker.cs [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+                }
+
+                return path;
+            }
+
+            try
+            {
+                bool GetPath(uint bufferSize, out string outputPath)
+                {
+                    var charArray = new char[bufferSize];
```
Comment (ayende) [inline src/Sparrow/Utils/DiskSpaceChecker.cs:185]: Can we use `ArrayPool.Rent` ? This is called quite frequently, as I recall

Comment (grisha-kotler) [reply src/Sparrow/Utils/DiskSpaceChecker.cs:185]: It's called only once for each storage environment.
The drive/mount point won't change while it's open.

**Review by ppekrol — APPROVED** [review]

File: src/Sparrow/Utils/DiskSpaceChecker.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Sparrow/Utils/DiskSpaceChecker.cs:35]: done

**Comment by ppekrol** [comment]
Can you apply this to v4.1?

**Pull request merged**